client/src/components/DataFiles/DataFilesSidebar/DataFilesSidebar.jsx

```diff
@@ -57,77 +59,88 @@ const DataFilesSidebar = ({ readOnly }) => {
+    shallowEqual
+  );
+
+  const match = useRouteMatch();
+
+  var sidebarItems = [];
+
+  systems.forEach(sys => {
+    sidebarItems.push({
+      to: `${match.path}/${sys.api}/${sys.scheme}/${sys.system ? `${sys.system}/` : ''}`,
+      label: sys.name,
+      iconName: sys.icon || 'my-data',
+      disabled: false,
+      hidden: false,
+    })
+  })
+
+  const addItems = [
+    {
+      className: styles['add-button-item'],
+      children: <DataFilesAddButton readOnly={readOnly}/>,
+    }
+  ]
+
   return (
     <div className={styles['root']}>
-      <div className="data-files-sidebar">
-        <div id="add-button-wrapper">
-          <ButtonDropdown isOpen={dropdownOpen} toggle={toggleDropdown}>
-            <DropdownToggle
-              color="primary"
-              id="data-files-add"
-              className="data-files-btn"
-            >
-              + Add
-            </DropdownToggle>
-            <DropdownMenu>
-              <DropdownItem
-                className={styles[writeItemStyle]}
-                onClick={toggleMkdirModal}
-                disabled={disabled}
-              >
-                <i className="icon-folder" /> Folder
-              </DropdownItem>
-              {sharedWorkspaces && !sharedWorkspaces.readOnly && (
-                <DropdownItem onClick={toggleAddProjectModal}>
-                  <i className="icon-folder" /> Shared Workspace
-                </DropdownItem>
-              )}
-              <DropdownItem
-                className={`complex-dropdown-item ${styles[writeItemStyle]}`}
-                onClick={toggleUploadModal}
-                disabled={disabled}
-              >
-                <i className={`icon-upload`} />
-                <span className="multiline-menu-item-wrapper">
-                  Upload
-                  <small> Up to 500mb </small>
-                </span>
-              </DropdownItem>
-            </DropdownMenu>
-          </ButtonDropdown>
-        </div>
-        <div className="data-files-nav">
-          <Nav vertical>
-            {systems
-              ? systems.map((sys) => (
-                  <NavItem key={`${sys.name}`}>
-                    <NavLink
-                      tag={RRNavLink}
-                      to={`${match.path}/${sys.api}/${sys.scheme}/${
-                        sys.system ? `${sys.system}/` : ''
-                      }`}
-                      activeClassName="active"
-                    >
-                      <div className="nav-content">
-                        <Icon name={sys.icon || 'my-data'} />
-                        <span className="nav-text">{sys.name}</span>
-                      </div>
-                    </NavLink>
-                  </NavItem>
-                ))
-              : null}
-          </Nav>
-        </div>
-      </div>
+      <CommonSidebar
+        sidebarItems={sidebarItems}
+        addItemsBefore={addItems}
+        backgroundColor="white"
```

Comment:
**Concern.** The Portal UI background is already white, isn't it? This looks like an attempt to add white on white. I admit I haven't run the code, so please clarify what I am missing.

**Opportunity to Future-Proof the Code.**
<details><summary><strong>If</strong> the background <em>should</em> be set...</summary>

...<strong>then</strong> consider how this parameter can become semantically inaccurate.

Imagine we skin the background for one client to blue, or a (future) dark theme makes the background dark gray, but the code still says `backgroundColor="white"`. Not a disaster, but it is a disservice to future developers to name parameters something that could too easily, accidentally become inaccurate.

Can you find a way to pass the information to the Sidebar CSS module stylesheet, such that only CSS is aware of the background being set and to what color? Perhaps a parameter name that expresses the state of the Sidebar that leads to the need to make the background white.

</details>

Reply:
The Sidebar by default has a gray background. See [here](https://github.com/TACC/Core-Portal/blob/task/FP-1560-use-common-sidebar/client/src/components/_common/Sidebar/Sidebar.module.css). We would need to override that in order to change the background color. There are instances e.g in the Applications page where we need a gray sidebar as well as a white sidebar so we do need some way to tell the Sidebar component what color to use. Can look into some alternative ways.

Reply:
Oh, I remember this now. [I knew this was gonna come up.](https://github.com/TACC/Core-Portal/pull/610#pullrequestreview-914930643) Yes, please do, thank you.

Reply:
Now that I know what's going on, my gut says a simpler solution (than "If the background should be set...") is transparent bkgd always, and let the "gray use-case" React components add the unique background.

Reply:
> Now that I know what's going on, my gut says a simpler solution (than "If the background should be set...") is have transparent background by default, and let the "gray use-case" React components add the unique background.

Interesting. That could be a good compromise